src/angleaxis_types.jl

```diff
@@ -10,24 +10,38 @@ arbitrary axis `[x, y, z]`.
 Note that the axis is not unique for θ = 0, and that this parameterization does
 not continuously map the neighbourhood of the null rotation (and therefore
 might not be suitable for autodifferentation and optimization purposes).
+
+Note: by default, the constructor will renormalize the input so that the axis
+has length 1 (x² + y² + z² = 1).
+
+Renormalization can be skipped by passing `Val{false}()` as an additional constructor
+argument, in which case the user provides the guarantee that the input arguments
+represent a normalized rotation axis. Operations on an `AngleAxis` with a rotation
+axis that does not have unit norm, created by skipping renormalization in this fashion,
+are not guaranteed to do anything sensible.
 """
 struct AngleAxis{T} <: Rotation{3,T}
     theta::T
     axis_x::T
     axis_y::T
     axis_z::T
 
-    # Ensure axis is normalized
-    function AngleAxis{T}(θ, x, y, z) where T
-        norm = sqrt(x*x + y*y + z*z)
-        # Not sure what to do with theta?? Should it become theta * norm ?
-        new(θ, x/norm, y/norm, z/norm)
+    @inline function AngleAxis{T}(θ, x, y, z, ::Val{Normalize} = Val{true}()) where {T, Normalize}
```

Comment:
Good call; just tested: LLVM is indeed able to optimize that branch away. I'll update the PR.